hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -126,16 +143,17 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     boolean acquiredLock = false;
     IOException exception = null;
     OMClientResponse omClientResponse = null;
+    Result result = null;
     try {
       // check Acl
       checkKeyAcls(ozoneManager, volumeName, bucketName, keyName,
           IAccessAuthorizer.ACLType.CREATE, OzoneObj.ResourceType.KEY);
 
       // Check if this is the root of the filesystem.
       if (keyName.length() == 0) {
-        return new OMDirectoryCreateResponse(null,
-            omResponse.setCreateDirectoryResponse(
-                CreateDirectoryResponse.newBuilder()).build());
+        throw new OMException("Directory create failed. Cannot create " +
```

Comment:
Is this the expected behavior, as the original code has returned normal OMResponse.

Reply:
I thought the previous response was wrong. We should return the correct error code, right?

Reply:
Yeah the change seems correct to me. @bharatviswa504 what do you think?

Reply:
Looks like hdfs does the same

Tried on a HDFS cluster.
[root@ozone-test-bh-2 ~]# hdfs dfs -mkdir hdfs://ozone-test-bh-6.vpc.cloudera.com:8020
mkdir: Call From ozone-test-bh-2.vpc.cloudera.com/10.65.51.138 to ozone-test-bh-6.vpc.cloudera.com:8020 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused
[root@ozone-test-bh-2 ~]# hdfs dfs -mkdir hdfs://ozone-test-bh-2.vpc.cloudera.com:8020
mkdir: `Item: hdfs://ozone-test-bh-2.vpc.cloudera.com:8020 parent's path is null. This can happen if mkdir is called on root, so there's no parent.': No such file or directory
[root@ozone-test-bh-2 ~]# hdfs dfs -mkdir hdfs://ozone-test-bh-2.vpc.cloudera.com:8020/
mkdir: `hdfs://ozone-test-bh-2.vpc.cloudera.com:8020/': File exists

Reply:
This change breaks Contract tests too.
So instead of throwing an exception, I am returning a Result code. Currently, the createDirectory client API returns a void. And added a TODO to return a boolean instead and then we can pass false when directory already exists.